src/edu/csus/ecs/pc2/ui/SubmissionSolutionList.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import edu.csus.ecs.pc2.imports.ccs.IContestLoader;
+
+/**
+ * List of judge's solutions
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class SubmissionSolutionList extends ArrayList<SubmissionSampleLocation> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 6231245379952020474L;
+	
+	public SubmissionSolutionList() {
+		super();
+	}
+	
+	public SubmissionSolutionList(File cdpPath) {
+		super();
+		loadCDPSampleSubmissionList(cdpPath);
+	}
+	
+
+    /**
+     * get all directories and child directories (recurses).
+     * 
+     * @param directory
+     * @return list of directory names
+     */
+	// TODO REFACTOR move to FileUtilities.getAllDirectoryEntries
+    public static List<String> getAllDirectoryEntries(String directory) {
+
+        ArrayList<String> list = new ArrayList<>();
+
+        File[] files = new File(directory).listFiles();
+        
+        if (files != null) {
+        	
+        	for (File entry : files) {
+        		if (entry.isDirectory()) {
+        			list.add(directory + File.separator + entry.getName());
+        			if (!(entry.getName().equals(".") || entry.getName().equals(".."))) {
+        				list.addAll(getAllDirectoryEntries(directory + File.separator + entry.getName()));
+        			}
+        		}
+        	}
+        }
+
+        return list;
+    }
+    
+
+
+	/**
+	 * Load Judge's samples judgement types
+	 * @param cdpPath
+	 * @throws IOException 
+	 */
+    // TODO NOW use FileUtilities.getAllDirectoryEntries
+	private void loadCDPSampleSubmissionList(File cdpPath)  {
+		
+		SortedMap<SubmissionSampleLocation, String> uniqItems = new TreeMap<SubmissionSampleLocation, String>();
```

Comment:
It would sure help those of us trying to validate/approve this PR (not to mention those who will look at this code later) if there was some kind of comment explaining the purpose/function of the `SortedMap` declared here.  As it is, I'm left to try to figure out, for example, what the purpose/usage of the `String` in the `SortedMap` is intended to be used for.  (Yes, I can eventually probably figure it out -- but if the author would provide just a line or two of explanation of the purpose of the Map it would certainly make things easier [and make reviewing go faster]...)